feat: optimize output structure

hirasso opened the pull request:
Fixes #17 

### Description

Two output optimizations:

- Makes sure injecting the classes `shiki-dark`/`shiki-light` is working as intended. The `replace()` was wrongly assuming a trailing quote `"` after the first class name `shiki` and hence not injecting the classes at all
- Prepend code blocks with light/dark theme with a `<pre hidden></pre>` as discussed in #17

hirasso committed:
Make sure "shiki-dark" and "shiki-light" make it into the `<pre>`-classList

hirasso committed:
Prevent markdown-it from auto-wrapping the `div.shiki-container`

antfu commented on src/index.ts:
```diff
@@ -119,10 +119,10 @@ const MarkdownItShiki: MarkdownIt.PluginWithOptions<Options> = (markdownit, opti
-        .replace('<pre class="shiki"', '<pre class="shiki shiki-light"')
-      return `<div class="shiki-container language-${lang}">${dark}${light}</div>`
+        .replace('<pre class="shiki ', '<pre class="shiki shiki-light ')
+      return `<pre hidden></pre><div class="shiki-container language-${lang}">${dark}${light}</div>`
```
Can we have a comment here for future reference? Thanks

hirasso replied:
Sure! In the HTML or in the source code, what du you prefer?

antfu replied:
In the source, thanks!

hirasso committed:
Explain the the reasoning behind the `<pre hidden>` element at the beginning of the output

hirasso committed:
Optimize `<pre hidden>` explanation

Using the same wording as on markdown-it's documentation site

@see https://markdown-it.github.io/markdown-it/

hirasso commented:
Thank you! 🙏
